hummingbot/strategy/cross_exchange_market_making/cross_exchange_market_making.pyx

```diff
@@ -520,25 +569,36 @@ cdef class CrossExchangeMarketMakingStrategy(StrategyBase):
 
             # Call c_check_and_hedge_orders() to emit the orders on the taker side.
             try:
-                self.c_check_and_hedge_orders(market_pair)
+                await self.check_and_hedge_orders(market_pair)
             except Exception:
                 self.log_with_clock(logging.ERROR, "Unexpected error.", exc_info=True)
 
-    cdef c_did_complete_buy_order(self, object order_completed_event):
+    def did_fill_order(self, order_filled_event):
+        if self._hedge_filled_maker_order_task is None or self._hedge_filled_maker_order_task.done():
+            if not self._has_unhedged_market_fill:
+                # Maker order filled, submit taker hedge order
+                self._has_unhedged_market_fill = True
+                self._hedge_filled_maker_order_task = safe_ensure_future(
+                    self.hedge_filled_maker_order(order_filled_event)
+                )
+            else:
+                # Maker order filled, Taker hedge order filled
+                self._has_unhedged_market_fill = False
+                # Cancell all leftover active orders
+                for _, limit_order in self.active_limit_orders:
+                    self.cancel_order(None, limit_order.client_order_id)
```

Comment:
I think it is necessary to check if the fill order event is not for a taker order. If it is for a taker order there is no need to do any other calculation, since the `self._has_unhedged_market_fill` condition will not need to change

Reply:
`_has_unhedged_market_fill` is True only if a maker order was newly filled, which triggers a submission of a taker order. If a taker order is filled, `_has_unhedged_market_fill` == True will indicate already that it is a taker order that was filled.